spinalcordtoolbox/reports/qc.py

```diff
@@ -217,35 +217,23 @@ def smooth_centerline(self, mask, ax):
         ax.get_xaxis().set_visible(False)
         ax.get_yaxis().set_visible(False)
 
-    def __call__(self, func):
-        """wrapped function (f).
+    def layout(self, qcslice_layout, qcslice_type):
+        """The main entry point for actually *using* a QcImage instance."""
+        self.qc_report.qc_params.orientation = qcslice_type.get_name()
```

Comment:
Honestly, that commit was my main motivation for this whole cleanup! That code complexity was completely gratuitous.

In case you're curious, here's a sketch of what was going on:
```py3
class QcImage:
    def __init__(self, qc_report, interpolation, ...):
        self.qc_report = qc_report
        ...

    def __call__(self, func):
        def wrapped_f(sct_slice):
            ...
            self.qc_report.make_content_path()
            ...
            stuff = func(sct_slice)
            ...
        return wrapped_f

qcslice_type = qcslice.Axial(...)
def qcslice_layout(x): return x.mosaic()

@QcImage(qc_report, interpolation, ...)
def layout(y):
    return qcslice_layout(y)

layout(qcslice_type)
```

The sequence of events for `@QcImage` and the following lines is roughly:
```py3
# Make a blank instance of the class.
val_1 = QcImage.__new__(QcImage)

# Initialize the new instance.
val_2 = QcImage.__init__(val_1, qc_report, interpolation, ...)

# Temporarily define the layout function.
def val_3(y):
    return qcslice_layout(y)

# Pass it through the decorator.
# What happens here would be:
#   layout = val_2(val_3)
# Which is implemented as:
#   layout = QcImage.__call__(val_2, val_3)
# But there's a nested function, so let's go through the steps in more detail:
def wrapped_f(sct_slice):
    ...
    val_2.qc_report.make_content_path()
    ...
    stuff = val_3(sct_slice)
    ...
layout = wrapped_f

# Finally, call it!
layout(qcslice_type)
```

Aside from all the low-level Python knowledge needed to unravel this, figuring out which variables are in scope for which lines of code was really tricky.